api/cluster/controller.go

```diff
@@ -327,12 +328,15 @@ func (c *controller) Delete(ctx context.Context, modelService *models.Service) (
 		}
 	}
 
-	if modelService.RevisionID > 1 {
-		vsName := fmt.Sprintf("%s-%s-%s", modelService.ModelName, modelService.ModelVersion, models.VirtualServiceComponentType)
-		if err := c.deleteVirtualService(ctx, vsName, modelService.Namespace); err != nil {
-			log.Errorf("unable to delete virtual service %v", err)
-			return nil, ErrUnableToDeleteVirtualService
-		}
+	vsCfg, err := NewVirtualService(modelService, "")
```

Comment:
when does modelService.RevisionID <= 0 happen? or was the check useless to begin with?

Reply:
When we haven't introduced model version revision features (prior to Merlin 0.34.0).

Yeah, the check was faulty resulting in Merlin not deleting:
1. VS of version endpoints without revision (the ones deployed by Merlin<0.34.0)
2. VS of version endpoints with revision 1 (the first deployment of version endpoint)
And will only delete VS of version endpoints with revision > 1.